llms/mlx_lm/models/openelm.py

```diff
@@ -0,0 +1,247 @@
+from dataclasses import dataclass
+from typing import Dict, List, Optional, Tuple, Union
+
+import mlx.core as mx
+import mlx.nn as nn
+
+from .base import BaseModelArgs
+
+
+@dataclass
+class ModelArgs(BaseModelArgs):
+    model_type: str
+    head_dim: int
+    num_transformer_layers: int
+    model_dim: int
+    vocab_size: int
+    ffn_dim_divisor: int
+    num_query_heads: List
+    num_kv_heads: List
+    ffn_multipliers: List
+    ffn_with_glu: bool = True
+    normalize_qk_projections: bool = True
+    share_input_output_layers: bool = True
+    rms_norm_eps: float = 1e-6
+    rope_theta: float = 10000
+    rope_traditional: bool = False
+    rope_scaling: Optional[Dict[str, Union[float, str]]] = None
+
+    def __post_init__(self):
+        if self.rope_scaling:
+            required_keys = {"long_factor", "type"}
+            if not all(key in self.rope_scaling for key in required_keys):
+                raise ValueError(f"rope_scaling must contain keys {required_keys}")
+
+            if self.rope_scaling["type"] != "linear":
+                print(
+                    "[WARNING] rope_scaling 'type' currently only supports 'linear' setting rope scaling to false."
+                )
```

Comment:
What is the purpose of this? I don't think this model uses rope scaling, so could you remove the stuff on rope scaling and just default it to 1.0?

Reply:
Yes, will do !

I thought the logic defaults to 1 anyways.

Reply:
Done ✅